### PR TITLE
Publishing APIs through API Manager

### PR DESCRIPTION
**Pull request by dinusha92**
### Purpose

Fixes: https://github.com/wso2/product-microgateway/issues/1594
### Issues

Fixes #

### Automation tests
 - Unit tests added: No
 - Integration tests added:  Pending because of APIM mocking

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/pkg/xds/xds_server.go [hunk 1]
```diff
@@ -266,7 +266,9 @@ func UpdateAPI(byteArr []byte, upstreamCerts []byte, apiType string) {
 		// due to any modifications to the code.
 		logger.LoggerXds.Info("API type is not cuurently supported by WSO2 micro-gateway")
 	}
-
+	//:TODO: since currently labels are not taking from x-wso2-label, I have made it to be taken from the method
+	// argument.
```
Comment (VirajSalaka) [inline adapter/pkg/xds/xds_server.go:270]: Shall we remove the new label assignment from the above code segment as well.

File: adapter/pkg/synchronizer/apis_fetcher.go [hunk 2]
```diff
@@ -214,3 +215,71 @@ func PushAPIProjects(payload []byte) error {
...
+			}
+			break
+		} else {
+			// Keep the iteration still until all the envrionment response properly.
+			logger.LoggerMgw.Errorf("Error occurred while fetching data from control plane: %v", data.Err)
+			go func(d SyncAPIResponse) {
+				// Retry fetching from control plane after a configured time interval
+				logger.LoggerMgw.Debugf("Time Duration for retrying: %v", 5*time.Second)
+				time.Sleep(5 * time.Second)
+				logger.LoggerMgw.Infof("Retrying to fetch API data from control plane.")
```
Comment (VirajSalaka) [inline adapter/pkg/synchronizer/apis_fetcher.go:279]: Infof with no variables assigned.
I wonder how this not identified from go vet

Comment (dinusha92) [reply adapter/pkg/synchronizer/apis_fetcher.go]: Fixed with https://github.com/wso2/product-microgateway/pull/1595/commits/66f646dd975eb98c2efc9f6a7ce2d2a43391c0d7

File: adapter/pkg/xds/xds_server.go (hunk 1 above)
Comment (dinusha92) [reply adapter/pkg/xds/xds_server.go:272]: Fixed with https://github.com/wso2/product-microgateway/pull/1595/commits/66f646dd975eb98c2efc9f6a7ce2d2a43391c0d7

**Review by VirajSalaka — APPROVED** [review]

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**